pre-commit.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# Run this script before you commit.
+
+# TODO: merge into one once we have Python 3.
+
+if ! [ -e venv2.7 ]; then
+  virtualenv -p python2.7 venv2.7 || exit 1
+  venv2.7/bin/pip install -r requirements-server.txt || exit 1
+  venv2.7/bin/pip install -r requirements.docs.txt || exit 1
+  rm -rf worker/codalabworker.egg-info  # Need to clear because of different Python versions
+  venv2.7/bin/pip install -e worker  # Not sure why this is necessary
+  venv2.7/bin/pip install -e .
```

Comment:
Why are we installing codalab etc. before each commit? (it seems like the main purpose of ./pre-commit.sh might be just to fix style and generate docs?)

Reply:
Generating docs requires having a running version of CodaLab (since it needs to load all the libraries).